lib/spack/docs/build_systems/autotoolspackage.rst

```diff
@@ -112,20 +112,45 @@ phase runs:
 
 .. code-block:: console
 
-   $ libtoolize
-   $ aclocal
-   $ autoreconf --install --verbose --force
+   $ autoreconf --install --verbose --force -I <aclocal-prefix>/share/aclocal
+
+In case you need to add more arguments, override ``autoreconf_extra_args``
+in your ``package.py`` on class scope like this:
+
+.. code-block:: python
+
+   autoreconf_extra_args = ["-Im4"]
 
 All you need to do is add a few Autotools dependencies to the package.
 Most stable releases will come with a ``configure`` script, but if you
-check out a commit from the ``develop`` branch, you would want to add:
+check out a commit from the ``master`` branch, you would want to add:
 
 .. code-block:: python
 
-   depends_on('autoconf', type='build', when='@develop')
-   depends_on('automake', type='build', when='@develop')
-   depends_on('libtool',  type='build', when='@develop')
-   depends_on('m4',       type='build', when='@develop')
+   depends_on('autoconf', type='build', when='@master')
+   depends_on('automake', type='build', when='@master')
+   depends_on('libtool',  type='build', when='@master')
+
+When the package has ``autoconf`` as dependenency, and ``m4`` is only used
+as the script interpreter of ``autoconf`` and not for other purposes, it
+does not have to be listed explicitly and can be removedd as it is not needed.
```

Comment:
This sentence is still here though, I'm not very attached to my opinion about removing it. But in this sentence it's somwhat ambiguous what "it" refers to, m4 or autoconf. How about:

```suggestion
It is typically redundant to list the ``m4`` macro processor package as a
dependency, since ``autoconf`` already depends on it.
```

Reply:
@haampie  Applied your suggestion, thanks!

It is typically redundant to list the ``m4`` macro processor package as a
dependency, since ``autoconf`` already depends on it.